Tools/LambdaTestTool-v2/tests/Amazon.Lambda.TestTool.IntegrationTests/ApiGatewayResponseExtensionsJsonInference.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Amazon.ApiGatewayV2;
+using Amazon.Lambda;
+using Amazon.IdentityManagement;
+using Xunit;
+using Amazon.Lambda.APIGatewayEvents;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Amazon.Lambda.TestTool.Extensions;
+
+namespace Amazon.Lambda.TestTool.IntegrationTests
+{
+    public class ApiGatewayResponseExtensionsJsonInference : IAsyncLifetime
+    {
+        private readonly ApiGatewayTestHelper _helper;
+        private string _httpApiV2Id;
+        private string _lambdaArn;
+        private string _httpApiV2Url;
+        private string _roleArn;
+        private readonly HttpClient _httpClient;
+
+        public ApiGatewayResponseExtensionsJsonInference()
+        {
+            var apiGatewayV2Client = new AmazonApiGatewayV2Client(RegionEndpoint.USWest2);
+            var lambdaClient = new AmazonLambdaClient(RegionEndpoint.USWest2);
+            var iamClient = new AmazonIdentityManagementServiceClient(RegionEndpoint.USWest2);
+
+            _helper = new ApiGatewayTestHelper(null, apiGatewayV2Client, lambdaClient, iamClient);
+            _httpClient = new HttpClient();
+        }
+
+        public async Task InitializeAsync()
+        {
+            // Create IAM Role for Lambda
+            _roleArn = await _helper.CreateIamRoleAsync();
+
+            // Create Lambda function
+            var lambdaCode = @"
+            exports.handler = async (event, context, callback) => {
+                console.log(event);
+                callback(null, event.body);
+            };";
+            _lambdaArn = await _helper.CreateLambdaFunctionAsync(_roleArn, lambdaCode);
+
+            // Create HTTP API (v2)
+            (_httpApiV2Id, _httpApiV2Url) = await _helper.CreateHttpApi(_lambdaArn, "2.0");
+
+            // Wait for the API Gateway to propagate
+            await Task.Delay(10000); // Wait for 10 seconds
```

Comment:
Instead of a sleep can we ping the endpoint till we get a Http StatusCode 200 or it has been 10 seconds?

Reply:
fixed in latest revision